rust/actyx/ax/src/cmd/settings/local/init.rs

```diff
@@ -0,0 +1,79 @@
+use crate::cmd::{determine_ax_default_data_dir, AxCliCommand};
+use ax_core::{
+    node::{self, lock_working_dir},
+    settings::Database,
+    util::formats::{ActyxOSCode, ActyxOSError, ActyxOSResult},
+};
+use futures::{stream, Stream};
+use serde::Serialize;
+use std::path::PathBuf;
+
+#[derive(Serialize)]
+pub struct SettingLocalInitResult {
+    created: bool,
+}
+
+#[derive(clap::Parser, Debug, Clone)]
+pub struct SettingsLocalInitOpts {
+    /// Enable to fail the command when the assigned WORKING_DIR already has
+    /// settings.db initialized within
+    #[arg(long)]
+    fail_on_existing: bool,
+    /// Path where to store all the data of the Actyx node.
+    #[arg(
+        long,
+        env = "ACTYX_PATH",
+        long_help = "Path where to store all the data of the Actyx node. \
+          Defaults to creating <current working dir>/actyx-data"
+    )]
+    working_dir: Option<PathBuf>,
+}
+
+pub struct SettingsLocalInit();
+impl AxCliCommand for SettingsLocalInit {
+    type Opt = SettingsLocalInitOpts;
+    type Output = SettingLocalInitResult;
+    fn run(opts: Self::Opt) -> Box<dyn Stream<Item = ActyxOSResult<Self::Output>> + Unpin> {
+        let r = Box::pin(run(opts));
+        Box::new(stream::once(r))
```

Comment:
What is the difference between `Box::new` and `stream::once(r).boxed()` asides from the `Pin<...>` on the latter?

Reply:
Not sure if there's any.

Reply:
I was under the impression that it needed to be pinned. @rkuhn am I missing something?

Reply:
It only needs to be pinned if the generator is `!Unpin`, which rustc would complain about.